steps/12_proba_forecasting/step.md

```diff
@@ -0,0 +1,89 @@
+
+* `predict_interval` example: if `y` has column names `"foo"`, `"bar"`, and `alpha=0.05` the return has column indices `("foo", 0.05)`, `("foo", 0.5)`, `("foo", 0.95)`, `("bar", 0.05)`,`("bar", 0.5)`, and `("bar", 0.95)`.
+
+* `predict_proba(self, fh=None, X=None, joint=False) -> tensorflow-probability.Distribution` returns a `tensorflow-probability` `Distribution` object. If `joint=False`, it is a vectorized distribution with one vector dimension, elements corresponding to elements of `fh`, in same sequence. If `joint=True`, it is a single joint distribution over a `len(fh)`-dimensional domain. Marginals of the `joint=True` return must be identical with the `joint=False` case.
+
+
+### base class defaulting
+
+If `predict_proba` is implemented, the other three `predict` functions default to calls in the following way:
+
+* `predict` calls `predict_proba` and then the mean
+* `predict_var` calls `predict_proba` and then the (co-)variance
+* `predict_interval` calls `predict_proba` and then the quantile function
+
+(of the distribution object returned by `predict_proba`)
+ 
+### new tags
+
+Each estimator may or may not have the capability to predict variances, intervals, etc. For this, the following boolean tags are introduced, with their obvious meaning:
+
+* `capability:predict_var`
+* `capability:predict_interval`
+* `capability:predict_proba`
+
+### public/private interface
+
+Each new method has a private counterpart which is called internally, i.e., `_predict_var`, `_predict_interval`, `_predict_proba`. As in the generic `sktime` design, the public method contains "plumbing", while the private class is the extension locus and contains only method logic.
+
+### Downwards compatibility: `predict`
+
+To ensure downwards compatibility, the `return_pred_int` and `alpha` parameters will still be accepted by `predict` until the end of the next deprecation cycle.
+
+If these are passed and `return_pred_int=True`, the base class directs the `predict` call to `predict_interval` where possible; for downwards compatibility, a temporary boolean tag `old_predict_interval_logic` directs to `_predict` instead, with the aim to remove prediction interval 
```

Comment:
```suggestion
To ensure downwards compatibility, the `return_pred_int` and `alpha` parameters will still be accepted by `predict` until the end of the next deprecation cycle.

If these are passed and `return_pred_int=True`, the base class directs the `predict` call to `predict_interval` where possible; Note: this is always possible once a specific forecaster is refactored. Whether a forecaster is already refactored will be checked for in an additional class method `_has_refactored_predict_interval()`;

Since the return type of prediction intervals also changes, a boolean tag `keep_old_return_type=True` will temporarily be passed to `predict`, allowing the user to decide whether he wants to use the old return type (also possible until the next deprecation cycle". 
```

Reply:
makes sense, thanks!